From 622c629ca9ceacb3e3bfededfdf0074d16d75eec Mon Sep 17 00:00:00 2001
From: Ajit Khaparde <ajit.khaparde@broadcom.com>
Date: Mon, 30 Oct 2017 11:40:16 -0600
Subject: [PATCH] net/bnxt: fix vlan spoof configuration

We are not programming the vlan_spoof setting currently owing to
an unnecessary check. This patch fixes that.

Fixes: 36735a932ca7 ("net/bnxt: support set VF QOS and MAC anti spoof")
---
 drivers/net/bnxt/rte_pmd_bnxt.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/net/bnxt/rte_pmd_bnxt.c b/drivers/net/bnxt/rte_pmd_bnxt.c
index 2061466..d62ccc1 100644
--- a/drivers/net/bnxt/rte_pmd_bnxt.c
+++ b/drivers/net/bnxt/rte_pmd_bnxt.c
@@ -322,9 +322,6 @@ int rte_pmd_bnxt_set_vf_vlan_anti_spoof(uint8_t port, uint16_t vf, uint8_t on)
 	if (vf >= dev_info.max_vfs)
 		return -EINVAL;
 
-	if (on == bp->pf.vf_info[vf].vlan_spoof_en)
-		return 0;
-
 	rc = bnxt_hwrm_func_cfg_vf_set_vlan_anti_spoof(bp, vf, on);
 	if (!rc) {
 		bp->pf.vf_info[vf].vlan_spoof_en = on;
-- 
1.9.1

